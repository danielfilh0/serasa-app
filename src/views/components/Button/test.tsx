import { render, screen } from '@testing-library/react'

import { Button } from '.'

describe('<Button />', () => {
  it('should render', () => {
    const { container } = render(<Button>Button</Button>)

    expect(screen.getByRole('button', { name: /Button/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render disabled', () => {
    const { container } = render(<Button disabled>Button</Button>)

    expect(screen.getByRole('button', { name: /Button/i })).toHaveAttribute(
      'aria-disabled',
      'true'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
