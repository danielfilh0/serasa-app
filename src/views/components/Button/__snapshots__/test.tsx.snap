// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
<button
  aria-disabled="false"
  class="bg-magenta-accent text-light-solid w-full rounded-lg h-10 md:h-12 text-base text-center font-bold transition-colors border border-transparent hover:border-dark-high disabled:bg-gray-300 disabled:text-gray-600 disabled:border-transparent disabled:pointer-events-none"
>
  Button
</button>
`;

exports[`<Button /> should render disabled 1`] = `
<button
  aria-disabled="true"
  class="bg-magenta-accent text-light-solid w-full rounded-lg h-10 md:h-12 text-base text-center font-bold transition-colors border border-transparent hover:border-dark-high disabled:bg-gray-300 disabled:text-gray-600 disabled:border-transparent disabled:pointer-events-none"
  disabled=""
>
  Button
</button>
`;
